bdk_keychain/src/keychain_txout_index.rs

```diff
@@ -248,6 +250,61 @@ impl<K: Clone + Ord + Debug> KeychainTxOutIndex<K> {
         }
     }
 
+    /// Convenience method to call [`store_until_unused_count`] for all keychains.
+    ///
+    /// [`store_until_unused_count`]: Self::store_until_unused_count
+    pub fn store_all_until_unused_count(&mut self, unused_count: u32) -> bool {
+        let keychains = self.keychains.keys().cloned().collect::<Vec<_>>();
+        keychains
+            .into_iter()
+            .map(|keychain| self.store_until_unused_count(&keychain, unused_count))
+            .fold(false, |acc, v| acc || v)
+    }
+
+    /// Derives script pubkeys from the descriptor until the unused script count reaches
+    /// `unused_count`.
+    ///
+    /// Returns whether any new script pubkeys were derived.
+    pub fn store_until_unused_count(&mut self, keychain: &K, unused_count: u32) -> bool {
+        if unused_count < 1 {
+            return false;
+        }
+        let up_to = self
+            .last_active_index(keychain)
```

Comment:
I like the term `pad` for the naming.

Reply:
> We could add a way of enforcing this which could be handy i.e. derive_new returns an Option<(u32, &Script)> that is None when there are already too many unused ones. Out of scope for this PR though.

LMK if I am missing something, but it seems the `derive_next_unused` already maintaining this? Maybe we should call it `get_next_unused` because it only derives when there are no unused left. I think that solves this purpose, and we should not get a new unused when there are already unused left. And `derive_new` can stay as it is, as that's specifically asking to derive new keys.

Reply:
Added issue: https://github.com/LLFourn/bdk_core_staging/issues/110